Sephora-M/assignment3/LetterGrid.java

```diff
@@ -0,0 +1,117 @@
+package assignment3;
+
+import java.awt.Point;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Set;
+
+/**Grid of letters. Contains a method that, given a dictionnary represented as a Trie, allows to find 
+ * all the words that can be formed in the grid. 
+ * 
+ * @author Sephora-M
+ */
+public class LetterGrid {
+	char[][] grid;
+	int X;
+	int Y;
+	
+	public LetterGrid(char[][] grid) {
+		if (grid == null || grid[0].length == 0)
+			throw new IllegalArgumentException("Cannot instatiate an empty or a null grid.");
+		this.grid = grid;
+		Y = grid.length;
+		X = grid[0].length;
+	}
+	
+	/**
+	 * Given a Trie object, returns a set of all the words in the Trie that can be formed in the grid 
+	 * @param dict a Trie, the dictionary
+	 * @return set of all the words from the dictionary that can be from in the grid
+	 * 
+	 * Note: the idea is to start from each letter in the grid and perform a DFS-like search. To optimize
```

Comment:
nit: I'd put this between the method description and the `@param` line.

Reply:
ok :)